sdk/servicebus/Azure.Messaging.ServiceBus/tests/Receiver/ReceiverLiveTests.cs

```diff
@@ -63,12 +67,15 @@ await using (var scope = await ServiceBusScope.CreateWithQueue(enablePartitionin
                 var receivedMessageCount = 0;
                 var messageEnum = messages.GetEnumerator();
 
-                foreach (var item in await receiver.ReceiveBatchAsync(messageCount))
+                while (receivedMessageCount < messageCount)
                 {
-                    receivedMessageCount++;
-                    messageEnum.MoveNext();
-                    Assert.AreEqual(messageEnum.Current.MessageId, item.MessageId);
-                    Assert.AreEqual(item.DeliveryCount, 1);
+                    foreach (var item in await receiver.ReceiveBatchAsync(messageCount))
+                    {
+                        receivedMessageCount++;
+                        messageEnum.MoveNext();
+                        Assert.AreEqual(messageEnum.Current.MessageId, item.MessageId);
+                        Assert.AreEqual(item.DeliveryCount, 1);
+                    }
```

Comment:
I don't think this will work as expected because we are not completing the messages, so all the messages will stay in the queue.

Lets say first time we received 9 messages, it will set `receivedMessageCount` as 9. It will come again inside while loop and then foreach loop,  lets say this time we again received 9 messages then ` Assert.AreEqual(item.DeliveryCount, 1);` will fail because each time it will increase delivery count. And finally it will set `receivedMessageCount` as 18.  Now this time it will come outside of while loop but `Assert.AreEqual(messageCount, receivedMessageCount);` will also fail.

Reply:
Good point, we aren't actually completing the message in most of the places I updated.

However, the tests themselves do pass because of the lock timeout so we don't end up receiving the same messages again. Still, I don't think this is reliable, so I will update to do the final receive in ReceiveAndDelete mode.

Reply:
I think this is not the right solution and as you said this is not reliable for cases where we just receive messages in peek lock mode or abandon messages.

We just want to test the peeklock and abandon functionality, so how about we just send one message and receive that message for peek lock mode and abandon? this way our tests will always pass and also we can easily check `messageId`, `deliveryCount`..etc

Reply:
Even sending and receiving one message would suffer from the same underlying issue that receive may not receive the number of requested messages. I will go ahead and remove the assertions on the total counts for abandon/peek lock.

Reply:
Confirmed with @nemakam that if we request for one message, it is guaranteed that we will get one message back. Also, it is safe to rely on lock timeout so we don't end up receiving the same messages again for tests.

Reply:
Will go back to using the batch receive and relying on lock timeout.